test/unit/change_stream.test.ts

```diff
@@ -0,0 +1,256 @@
+import { Long, Timestamp } from 'bson';
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+
+import { ChangeStreamCursor } from '../../src/change_stream';
+import { MongoClient } from '../../src/mongo_client';
+import { MongoDBNamespace } from '../../src/utils';
+
+describe('class ChangeStreamCursor', function () {
+  afterEach(() => sinon.restore());
+  describe('get resumeOptions()', function () {
+    context('non-resume related options', function () {
```

Comment:
Definitely open to organizing the tests differently because they're a bit hard evaluate (although, the output from the tests is easy to understand).  I considered a table test but didn't know if that would be harder to understand than explicitly writing the tests.